File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    cnt = 0
    
    queue =  [(i,p) for i,p in enumerate(priorities)]
        

    while True:
        cur = queue.pop(0)
        if any(cur[1] < q[1] for q in queue):
            queue.append(cur)
            
        else:
            cnt += 1
            if cur[0] == location:
                return cnt

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const queue = priorities.map((priority, index) => ({priority, index}));
    let order = 0;
    
    while(queue.length > 0){
        const current = queue.shift();
        const top = queue.some(item => item.priority > current.priority)
        if(top){
            queue.push(current)
        }
        else{
            order++;
            if(current.index === location){
                return order
            }
        }
    }
    
}


